apps/communications/dialer/js/dialer.js

```diff
@@ -219,50 +186,47 @@ var CallHandler = (function callHandler() {
-      Contacts.findByNumber(number, function lookup(contact, matchingTel) {
-        LazyL10n.get(function localized(_) {
-          var title;
-          if (navigator.mozIccManager.iccIds.length > 1) {
-            title = _('missedCallMultiSims', {n: serviceId + 1});
-          } else {
-            title = _('missedCall');
-          }
+    Contacts.findByNumber(number, function lookup(contact, matchingTel) {
+      var _ = navigator.mozL10n.get;
+      var title;
+      if (navigator.mozIccManager.iccIds.length > 1) {
+        title = _('missedCallMultiSims', {n: serviceId + 1});
+      } else {
+        title = _('missedCall');
+      }
 
-          var body;
-          if (!number) {
-            body = _('from-withheld-number');
-          } else if (contact) {
-            var primaryInfo = Utils.getPhoneNumberPrimaryInfo(matchingTel,
-              contact);
-            if (primaryInfo) {
-              if (primaryInfo !== matchingTel.value) {
-                // primaryInfo is an object here
-                body = _('from-contact', {contact: primaryInfo.toString()});
-              } else {
-                body = _('from-number', {number: primaryInfo});
-              }
-            } else {
-              body = _('from-withheld-number');
-            }
+      var body;
+      if (!number) {
+        body = _('from-withheld-number');
+      } else if (contact) {
+        var primaryInfo = Utils.getPhoneNumberPrimaryInfo(matchingTel,
+          contact);
+        if (primaryInfo) {
+          if (primaryInfo !== matchingTel.value) {
+            // primaryInfo is an object here
+            body = _('from-contact', {contact: primaryInfo.toString()});
           } else {
-            body = _('from-number', {number: number});
+            body = _('from-number', {number: primaryInfo});
           }
+        } else {
+          body = _('from-withheld-number');
+        }
+      } else {
+        body = _('from-number', {number: number});
+      }
 
-          navigator.mozApps.getSelf().onsuccess = function getSelfCB(evt) {
-            var app = evt.target.result;
-
-            var iconURL = NotificationHelper.getIconURI(app, 'dialer');
-            var clickCB = function() {
-              app.launch('dialer');
-              window.location.hash = '#call-log-view';
-            };
-            var notification =
-              new Notification(title, {body: body, icon: iconURL});
-            notification.addEventListener('click', clickCB);
-          };
-        });
-      });
+      navigator.mozApps.getSelf().onsuccess = function getSelfCB(evt) {
+        var app = evt.target.result;
+
+        var iconURL = NotificationHelper.getIconURI(app, 'dialer');
+        var clickCB = function() {
+          app.launch('dialer');
+          window.location.hash = '#call-log-view';
+        };
+        var notification =
+          new Notification(title, {body: body, icon: iconURL});
```

Comment:
Since you're working on this, you can move Notification to use NotificationHelper and then you will be able to pass L10n object (so: 'l10nId' or {id: 'l10nId', args: {}}) instead of using mozL10n.get here. This will remove the need to guard the code with mozL10n.once/ready.

See: https://bugzilla.mozilla.org/show_bug.cgi?id=1111841
And https://developer.mozilla.org/en-US/Firefox_OS/Developing_Gaia/localization_code_best_practices#Writing_APIs_that_operate_on_L10nIDs